features/commands.js

```diff
@@ -85,6 +85,32 @@ https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html`,
         });
       }
     },
+    {
+      words: [`!stateupdates`, `!su`],
+      help: `Explains the issue with state updates being asynchronous`,
+      handleMessage: msg => {
+        msg.channel.send({
+          embed: {
+            title:
+              "State Updates May Be Asynchronous",
+            type: "rich",
+            description: `Often times you run into an issue like this
+\`\`\`js
+const handleEvent = e => {
+  setState(e.target.value);
+  console.log(state);
+}
+\`\`\`
+where \`state\` is not the most up to date value when you log it. This is caused by state updates being asynchronous.
```

Comment:
So the title of this is 
> State Updates May Be Asynchronous

But the wording here is 

> This is caused by state updates being asynchronous.

That's pretty confusing. The former statement is just a mirror of the [official docs](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) title and makes it sound uncertain whether or not that state updates are asynchronous, which seems to contradict the latter statement. 

I'm not sure which is more appropriate here 😬 I would assume the official docs are *probably* more credible, but I'm not sure how to re-word the latter statement (and I would also have to re-word my explanations in the gist that's linked here).

Reply:
Update: so I thought about it some more and realized that the reason why state in function components is always asynchronous is because of closures. A function component can only capture the state variable at any given time, and for the remainder of that closure it will always have that value that was captured.

This is different in the behavior of class components, as it's not dealing with closures. `this.state` *could* technically be updated synchronously, depending on when react ultimately decides to update it.

It's usually not good practice to expect state updates in class components to happen synchronously though, so best practice is to always treat state updates as asynchronous.

That being said, the wording here is probably fine since the gist is explicitly referring to state updates within function components. I might add a blurb with the difference in how state updates are handled within class components to the gist, as to provide more clarity on the subject.